centreon/src/Core/Service/Infrastructure/Repository/DbWriteServiceActionLogRepository.php

```diff
@@ -0,0 +1,266 @@
+                throw new RepositoryException(sprintf('Cannot find service to delete (ID: %d).', $serviceId));
+            }
+            
+            $this->writeServiceRepository->delete($serviceId);
+
+            $actionLog = new ActionLog(
+                self::SERVICE_OBJECT_TYPE,
+                $serviceId,
+                $service->getName(),
+                ActionLog::ACTION_TYPE_DELETE,
+                $this->contact->getId()
+            );
+            $this->writeActionLogRepository->addAction($actionLog);
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+
+            throw $ex;
+        }
+    } 
+
+    /**
+     * @inheritDoc
+     */
+    public function deleteByIds(int ...$serviceIds): void
+    {
+        $failedDeletions = [];
+        foreach ($serviceIds as $serviceId) {
+            try {
+                $service = $this->readServiceRepository->findById($serviceId);
+                if ($service === null) {
+                    throw new RepositoryException('Cannot find service to delete');
+                }
+                
+                $this->writeServiceRepository->delete($serviceId);
+
+                $actionLog = new ActionLog(
+                    self::SERVICE_OBJECT_TYPE,
+                    $serviceId,
+                    $service->getName(),
+                    ActionLog::ACTION_TYPE_DELETE,
+                    $this->contact->getId()
+                );
+                $this->writeActionLogRepository->addAction($actionLog);
+            } catch (\Throwable $ex) {
+                $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+                $failedDeletions[] = $serviceId;
+            }
+        }
+
+        if ($failedDeletions !== []) {
+            throw new RepositoryException('Failed to delete services with IDs: ' . implode(',', $failedDeletions));
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function add(NewService $newService): int
+    {
+        try {
+            $serviceId = $this->writeServiceRepository->add($newService);
+            if ($serviceId === 0) {
+                throw new RepositoryException('Service ID cannot be 0');
+            }
+
+            $actionLog = new ActionLog(
+                self::SERVICE_OBJECT_TYPE,
+                $serviceId,
+                $newService->getName(),
+                ActionLog::ACTION_TYPE_ADD,
+                $this->contact->getId()
+            );
+            $actionLogId = $this->writeActionLogRepository->addAction($actionLog);
+            $actionLog->setId($actionLogId);
+
+            $details = $this->getServicePropertiesAsArray($newService);
+            $this->writeActionLogRepository->addActionDetails($actionLog, $details);
+
+            return $serviceId;
+        } catch (\Throwable $ex) {
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
```

Comment:
you can specify business information with $newService in the context of log to have more informations from the log

Reply:
Done